### PR TITLE
First draft of NA guide

### PR DESCRIPTION
**Pull request by CMPeng**
- Network Dataset and Network Analysis Layer (Part 1) 
- Find Routes (Part 2) - 2 modules
- Generate Service Area (Part 3)
- Find Closest Facility (Part 4)

*The PR only contains guides from 1 to 4, and parts 5-7 will be in the next PR*

- Generate Origin Destination Cost Matrix (Part 5) - 2 modules
- Solve Location Allocation (Part 6) - only the network.analysis module
- Vehicle Routing Problem Service (Part 7)

*Parts 8 and 9 are in internal archive on geosaurus*
https://github.com/ArcGIS/geosaurus/pull/2783

### CODE REVIEW

**Comment by AtmaMani** [comment]
For https://github.com/ArcGIS/geosaurus/issues/2416

**Comment by CMPeng** [comment]
Hey @AtmaMani 
The `re_ordered_stops_cities_fset.spatial_reference = stops_cities_fset.spatial_reference` line is added to part2 for the FeatureSet Spatial reference issue I mentioned this morning.

File: guide/11-performing-network-analyses/guide_to_network_analysis_part1.ipynb [hunk 1]
```diff
@@ -0,0 +1,280 @@
+{
```
Comment (AtmaMani) [inline guide/11-performing-network-analyses/guide_to_network_analysis_part1.ipynb]: <p>Line 1 - please use one of 2 forms of the product name - either "ArcGIS API for Python" or "Python API" to keep in consistent in the guide.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/528/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (AtmaMani) [inline guide/11-performing-network-analyses/guide_to_network_analysis_part1.ipynb]: <p>Sentence 1 - product name has to be corrected</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/528/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (AtmaMani) [inline guide/11-performing-network-analyses/guide_to_network_analysis_part1.ipynb]: <p>"In order to model transportation networks, Network Dataset<strong>s</strong> are created from source features.." -&gt; make Dataset plural</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/528/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (AtmaMani) [inline guide/11-performing-network-analyses/guide_to_network_analysis_part1.ipynb]: <p>Hyperlink for NA services (line 1) is broken.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/528/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (AtmaMani) [inline guide/11-performing-network-analyses/guide_to_network_analysis_part1.ipynb]: <p>"However, there are situations <em>that </em><strong><em>where</em></strong> users prefer to use the customized NA services instead." replace 'that' with 'where'</p><p> </p><p>" to use the customized NA services on enterprise. " - Capitalize Enterprise as it refers to a product.</p><p> </p><p>When talking about class / module names, enclose them in <code>code</code> formatting - for instance - <code>network.analysis.find_routes</code></p><p> </p><p>"<strong>And</strong> the differences between APIs in these two modules are: " -&gt; remove 'And' and start sentence with The</p><p> </p><p>The last bullet point - "In situations where the output Feature Collection is oversized,..." is valid. But you should emphasize (in the previous bullet) that when using <code>features.user_proximity</code>, the inputs and outputs are Feature Service Layers, whereas in <code>network.analysis</code>, the inputs and outputs are in-memory <code>FeatureCollections</code>. Then the last point gets easier to understand.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/528/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

**Comment by AtmaMani** [comment]
Well written @CMPeng. I am posting comments for each notebook separately. I just requested a few changes for NB 1

**Comment by CMPeng** [comment]
thanks @AtmaMani 
Part 1 is modified according to your suggestions, and I have updated the product name, dataset(s), and function/module quotations in parts 2-4 as well.

**Comment by CMPeng** [comment]
@AtmaMani 
I have addressed your revision suggestions for part 2, and done the same changes for comparison diff in parts 3 and 4. Thanks!

**Comment by AtmaMani** [comment]
<p>@cmpeng I reviewed part 2. It looks good. I requested for a few changes.</p>

**Comment by CMPeng** [comment]
Thanks again @AtmaMani  - 
The new comments are addressed for Part 2, and I changed the (1) extra parameters for web map saving, and (2) sdf extracted from result1['routes_layer'] directly for parts 3 and 4 as well.

**Comment by AtmaMani** [comment]
@CMPeng I reviewed part 3 and posted some comments. Can we discuss about the last part - cleaning feature collection output?

**Comment by CMPeng** [comment]
Hi @AtmaMani 
Requested changes have been made to Part 3, and I have also renamed the other notebooks. Thanks!

**Comment by AtmaMani** [comment]
@CMPeng I pushed some edits to the 4th notebook in this PR. Apart from some proofing edits, I simplified the viz of the routes, so the maps would look different. Could you take a look at the changes?

**Comment by CMPeng** [comment]
Thanks @AtmaMani  - I like the new visualizations. Looking good!

**Pull request merged**